src/accumulator/accumulated_map.rs

```diff
@@ -91,3 +94,30 @@ impl ops::BitOrAssign for InputAccumulatedValues {
         *self = *self | rhs;
     }
 }
+
+#[derive(Debug, Default)]
+pub struct AtomicInputAccumulatedValues(AtomicBool);
```

Comment:
so, this is a slightly nit-picking comment, but do you mind writing a unit test or for this one? the reason i ask is that my eyes kinda glaze over the contents and having some usage examples by the way of unit tests would be nice.